src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -791,13 +795,14 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
 
             if (mapWrapper.containsKey(pmt.Id)) {
                 alloWrapper pmtWrap = mapWrapper.get(pmt.Id);
-                Decimal amt = valueOrDefaultIfNull(pmt.npe01__Payment_Amount__c, 0);
-                //null or zero amount: delete all associated allocations
-                if (amt == 0) {
+                Decimal amt = pmt.npe01__Payment_Amount__c;
+                // null delete all associated allocations
```

Comment:
Seems like a good opportunity to abstract out some behavior. The method itself is a huge and could use cleanup, but for now I think it's safer to refactor what's being modified.
```
            Decimal amt = pmt.npe01__Payment_Amount__c;
            if (amt == null) {
                handleNullAmount(pmtWrap);
                continue;
            } else { 
```
```
/**
* @description If amount is null then delete all associated allocations
*/
private void handleNullAmount(alloWrapper pmtWrap) {
    dmlWrapper.objectsToDelete.addAll((List<SObject>)pmtWrap.listAllo);
    if (settings.Default_Allocations_Enabled__c && pmtWrap.defaultAllo != null) {
        dmlWrapper.objectsToDelete.add(pmtWrap.defaultAllo);
        dmlWrapper.objectsToDelete.addAll((List<SObject>)pmtWrap.defaultDupesById.values());
    }
}

Reply:
Yeah, I was hoping to eventually replace some of this with the service layer we are using via callable, but I can do that.

Reply:
I don't want it to do the continue statement, but will otherwise do as you suggest.  This is because I still want it to go through the default allocations block.  I agree this whole thing could be simplified, but I would rather refactor parts to share the service logic long term.